lib/catalog/data_driven_form_validator.rb

```diff
@@ -0,0 +1,86 @@
+module Catalog
+  class DataDrivenFormValidator
+    COMPONENTS = %i[text-field
+                    textarea-field
+                    field-array
+                    select-field
+                    checkbox
+                    sub-form
+                    radio
+                    tabs
+                    tab-item
+                    date-picker
+                    time-picker
+                    wizard
+                    switch-field
+                    plain-text].freeze
+
+    VALIDATORS = %i[required-validator
+                    min-length-validator
+                    max-length-validator
+                    exact-length-validator
+                    min-items-validator
+                    min-number-value
+                    max-number-value
+                    pattern-validator
+                    url-validator].freeze
+
+    DATA_TYPES = %i[integer
+                    float
+                    number
+                    boolean
+                    string].freeze
+
+    def self.valid_ddf?(ddf)
+      ddf = JSON.parse(ddf).with_indifferent_access
+      valid_fields?(ddf[:schema][:fields])
+
+      true
+    end
+
+    class << self
+      def valid_fields?(fields)
+        fields.each do |field|
+          valid_component?(field[:component].to_sym)
+          valid_data_type?(field[:dataType].to_sym) if field.key?(:dataType)
+          valid_validators?(field[:validate])
```

Comment:
Possibly a better name than `valid_validators`? Possibly pull out the `valid_` everywhere?  Though I don't mind it here as much in the physical class.

Reply:
`s/valid_//g`